src/components/Provider.js

```diff
@@ -20,7 +20,7 @@ function warnAboutReceivingStore() {
 
 export default class Provider extends Component {
   getChildContext() {
-    return { store: this.store }
+    return { store: this.store, react15CompatibilityMode: this.props.react15CompatibilityMode }
```

Comment:
But you shouldn't be blanket applying a perf-reducing setting when it may not apply to most connected components. Just make it default at the `connect()` level and remove the context stuff.